elasticdl/python/common/model_handler.py

```diff
@@ -61,57 +66,93 @@ def get_model_to_train(self, model):
         an elasticdl.layers.Embedding layer in ParameterServerStrategy.
         """
         if type(model) == tf.keras.Sequential or model._is_graph_network:
-            model = self._clone_model_for_sequential_and_functional(model)
+            model = self._replace_embedding_layer_to_clone_model(
+                model, tf.keras.layers.Embedding, Embedding
+            )
         else:
-            model = self._replace_embedding_attribute_for_subclass(model)
+            model = self._replace_embedding_attribute_for_subclass(
+                model, tf.keras.layers.Embedding, Embedding
+            )
         return model
 
-    def get_model_to_export(self, model, dataset):
+    def get_model_to_export(self, model, trained_params, dataset):
         """
-        To export model for tf-serving by tf.saved_model.save:
-        1. Add inputs and outputs to the model.
-        2. Restore Keras model and replace embedding parameters with trained
-        model.
+        Get the model which can be exported a SavedModel by
+        tf.saved_model.save.
         """
-        # TODO
-        pass
+        model = self._restore_keras_model_def(model)
+        if not model.inputs:
+            # build model to add inputs and outputs for tf-serving
+            model._build_model_with_inputs(inputs=dataset, targets=None)
+        for var in model.trainable_variables:
+            var.assign(trained_params[var.name])
+        return model
 
-    def _clone_model_for_sequential_and_functional(self, model):
+    def _restore_keras_model_def(self, model):
         """
-        Clone a new model with elasticdl.layers.Embedding for
-        Sequential and functional API model.
+        Restore Keras model definition by replacing elasticdl.layers.Embedding
+        layers with tf.keras.Embedding layers.
+        """
+        # clear keras model session to avoid clutter from old models/layers.
+        tf.keras.backend.clear_session()
+        if (
+            type(model) == tf.keras.models.Model
+            and not model._is_graph_network
+        ):
+            model = self._replace_embedding_attributes_for_subclass(
+                model, Embedding, tf.keras.layers.Embedding
+            )
+        else:
+            model = self._replace_embedding_layer_to_clone_model(
+                model, Embedding, tf.keras.layers.Embedding
+            )
+        return model
+
+    def _replace_embedding_layer_to_clone_model(
+        self, model, src_embedding_class, dst_embedding_class
+    ):
+        """
+        Clone a new model by cloning model and replace the
+        src_embedding_class layer with a dst_embedding_class.
         """
 
         def _clone_function(layer):
-            if type(layer) == tf.keras.layers.Embedding:
-                logger.info("Replace Keras Embedding with ElasticDL Embedding")
-                edl_embedding_layer = Embedding(
+            if type(layer) == src_embedding_class:
+                logger.info(
```

Comment:
maybe debug level log?

Reply:
Yes, users don't need to care about the log.